workloads/network-perf/compare.py

```diff
@@ -56,8 +83,12 @@ def compare(data):
                 proto]['message_size'][size]['num_threads'] :
                 result = data['test_type.keyword'][test_type]['protocol'][
                         proto]['message_size'][size]['num_threads'][thread]
-                comparison = compare(result[main_metric[test_type]])
+                if test_type == "rr" :
+                    comparison = compare(result[main_metric[test_type]], int(params.tolerance[0]), latency=True)
+                else:
+                    comparison = compare(result[main_metric[test_type]], tolerance=int(params.tolerance[0]))
                 if params.uuid[0] not in comparison :
-                    print("Test Failure : UPerf reports lower throughput for, Test: {}, Protocol {}, Message_size {}, Threads {}".format(
+                    print("Test Failure : UPerf reports regression for, Test: {}, Protocol {}, Message_size {}, Threads {}".format(
                         test_type,proto,size,thread))
                     exit(1)
```

Comment:
I suggest, lets capture the differences (for all message sizes and threads) and present them at the end. Versus exiting right after finding one difference. We should capture any possible regression.